src/Workspaces/Core/Portable/Workspace/Host/Metadata/IAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -35,30 +35,60 @@ internal interface IAnalyzerAssemblyLoaderProvider : IWorkspaceService
 /// </summary>
 internal abstract class AbstractAnalyzerAssemblyLoaderProvider : IAnalyzerAssemblyLoaderProvider
 {
-    private readonly ImmutableArray<IAnalyzerAssemblyResolver> _externalResolvers;
+#if NET
     private readonly Lazy<IAnalyzerAssemblyLoaderInternal> _shadowCopyLoader;
+    private readonly ImmutableArray<IAnalyzerAssemblyResolver> _assemblyResolvers;
 
-    public AbstractAnalyzerAssemblyLoaderProvider(IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+    public AbstractAnalyzerAssemblyLoaderProvider(IEnumerable<IAnalyzerAssemblyResolver> assemblyResolvers)
     {
-        _externalResolvers = [.. externalResolvers];
+        _assemblyResolvers = [.. assemblyResolvers];
         _shadowCopyLoader = new(CreateNewShadowCopyLoader);
```

Comment:
openly musing: i wonder if we get any value in deferring this.  esp. as the loader itself seems to defers doing any work until it is asked.

Reply:
I don't think that we get any benefit from this. At the same time though a few PRs further down this refactor we will need to be threading state through to the creation of the analyzer loader. 

One of the goals is that we get almost all knowledge of Razor out of workspaces and into the Razor specific layer. That means Razor will be providing a `IAnalyzerPathResolver` and `IAnalyzerAssemblyResolver` instance, via MEF, to this type. In order to correctly implement the path resolver though we need information from Visual Studio that requires accessing APIs that can't be done from a background thread. Having an explicit create function will help a bit here in getting that wired up. @jasonmalinowski and I have been discussing the best way to approach this.

Reply:
Yep, makes sense for me to keep it lazy for now. I imagine this will still look different eventually, but there's some indirection/laziness going to be present even in the new thing.